pkg/auth/providers/publicapi/login.go

```diff
@@ -76,6 +78,9 @@ func (h *loginHandler) login(actionName string, action *types.Action, request *t
 		if httperror.IsAPIError(err) {
```

Comment:
> Overall, I'm a bit uncomfortable with the error check being done below - near as I can tell this could affect all providers (so if I can trick any other provider into returning an error with this message I could lock people out of any provider). The line that I left this comment on seems to indicate that you could have activeDirectory login function return the appropriate HTTP error, which would limit the potential harm here. Would you mind elaborating on why you choose this other approach?

I actually like the approach of keeping this all in ad_provider or lower.  Updating to do it that way.